src/zenml/utils/source_utils.py

```diff
@@ -314,20 +314,44 @@ def import_class_by_path(class_path: str) -> Type[Any]:
     return getattr(mod, classname)  # type: ignore[no-any-return]
 
 
+class ModuleContextManager:
```

Comment:
Love the idea of the context manager, perfect tool for the job! I have one suggestion that could make it a little neater in my opinion, and that would be using the `@contextmanager` decorator instead of creating a separate class for it:

```python
from contextlib import contextmanager

@contextmanager
def prepend_python_path(path: str):
    try:
         sys.path.insert(0, path)
         yield
    finally:
        sys.path.remove(path)
```

The `yield` is essentially the end of `__enter__`, and the code inside the `finally` block will be executed after the context exited.

Reply:
Thank you for the tip! I have replaced the context manager.